apps/train_loc/lib/train_loc/manager.ex

```diff
@@ -86,23 +87,26 @@ defmodule TrainLoc.Manager do
     state = schedule_timeout(state)
 
     for event <- events, event.event == "put" do
+      Logger.debug(fn ->
+        "#{__MODULE__}: received event - #{inspect(event)}"
+      end)
+
       _ =
-        Logger.debug(fn ->
-          "#{__MODULE__}: received event - #{inspect(event)}"
-        end)
-
-      case ManagerEvent.from_string(event.data) do
-        {:ok, manager_event} ->
-          update_vehicles(manager_event, state)
-
-        {:error, reason} ->
-          _ =
-            Logger.error(fn ->
-              Logging.log_string("Manager Event Parsing Error", reason)
+        case BulkEvent.parse(event.data) do
```

Comment:
suggestion: this looks like it could be a `with`?
```elixir
with {:ok, raw_vehicles} <- BulkEvent.parse(event.data),
  Logger.info("Processing..."),
  feed = generate_feed(raw_vehicles, state),
  {:ok, _} <- @s3_api.put_object(...) do
    :ok 
  else
    {:error, :invalid_event, event} ->
      handle_old_event(event, state)
    ...
  end
```